src/NBomber/Domain/Errors.fs

```diff
@@ -15,12 +15,14 @@ type internal ValidationError =
     | TargetScenarioNotFound  of notFoundScenarios:string[] * registeredScenarios:string[]
     | DurationIsWrong         of scenarioNames:string[]
     | ConcurrentCopiesIsWrong of scenarioNames:string[]
+    
+    // ScenarioValidation errors
     | EmptyReportName 
     | EmptyScenarioName
     | DuplicateScenarioName of scenarioNames:string[]
     | EmptyStepName         of scenarioNames:string[]
-    | DuplicateStepName     of stepNames:string[]    
-    | AssertNotFound        of assertNumber:int * assertion:Assertion    
+    | DuplicateStepName     of stepNames:string[]  
+    | AssertionDoesntMatchScenario of scenarioNames:(ScenarioName *StepName)[]

```

Comment:
why not assert not found? for me personally it's simpler to think of... you specified assert which doesn't exist...